src/cephadm/cephadm.py

```diff
@@ -3194,7 +3194,7 @@ def command_shell(ctx):
             daemon_type = ctx.name
             daemon_id = None
     else:
-        daemon_type = 'osd'  # get the most mounts
+        daemon_type = 'shell'  # get limited set of mounts
```

Comment:
```
2024-04-24T16:57:49.540 DEBUG:teuthology.orchestra.run.smithi174:> sudo /home/ubuntu/cephtest/cephadm --image quay-quay-quay.apps.os.sepia.ceph.com/ceph-ci/ceph:5904d29475f5be602879d9fb26280e89b808d5cc shell -c /etc/ceph/ceph.conf -k /etc/ceph/ceph.client.admin.keyring --fsid 741fd43a-025b-11ef-bc93-c7b262605968 -- ceph orch apply mon '2;smithi137:172.21.15.137=smithi137;smithi174:172.21.15.174=smithi174'
2024-04-24T16:57:49.774 INFO:teuthology.orchestra.run.smithi174.stderr:Inferring config /etc/ceph/ceph.conf
2024-04-24T16:57:49.820 INFO:teuthology.orchestra.run.smithi174.stderr:Inferring config /etc/ceph/ceph.conf
2024-04-24T16:57:49.846 INFO:teuthology.orchestra.run.smithi174.stderr:Error: statfs /var/log/ceph/741fd43a-025b-11ef-bc93-c7b262605968: no such file or directory
```
we're still missing some mounts I guess

Reply:
current state with this patch
```root@vm-00 ~]# ls /var/log/ceph/
2faeb234-0270-11ef-ba7e-525400c013c5  cephadm.log
[root@vm-00 ~]# 
[root@vm-00 ~]# cephadm shell
Inferring fsid 2faeb234-0270-11ef-ba7e-525400c013c5
Inferring config /var/lib/ceph/2faeb234-0270-11ef-ba7e-525400c013c5/mon.vm-00/config
Using ceph image with id '46d143e249f3' and tag 'testing' created on 2024-04-24 19:22:00 +0000 UTC
quay.io/adk3798/ceph@sha256:ba17398e438b46bb849f229dc1f71473654ae2d59f0710775c3d76165a0e06fc
[ceph: root@vm-00 /]# ls /var/log/ceph/
ceph-client.ceph-exporter.vm-00.log  ceph-volume.log
```
so it seems like we're not getting the `/var/log/ceph/<fsid>` in the mounts

Reply:
Turns out this wasn't a missing mount. This happens if you try to run `cephadm shell`  with `--fsid <fsid>` with this patch before the log dir `/var/log/ceph/<fsid>` has been created (I think it appears when the first daemon wants to log there). I'm thinking we might just want to create this dir if we get here and it doesn't exist.